### PR TITLE
countDominantTaxa output table

### PR DESCRIPTION
**Pull request by himmil**

### CODE REVIEW

File: R/summaries.R
Comment (antagomir) [inline R/summaries.R]: If this is part of the function then provide the rounding digits as a function arguments (setting default to 3 is OK) so that users can ultimately choose as they like.

**Review by TuomasBorman — CHANGES_REQUESTED** [review]
round could be an integer --> that would simplify the code

How about using digits as parameter name? round() has a parameter called that.

Also add input check for round parameter

**Comment by antagomir** [comment]
We should use the same argument name than the `round` function itself, this will be more clear.

File: R/summaries.R
Comment (antagomir) [inline R/summaries.R]: If you do not give digits any default then I think it is not necessary to specify this argument, it can be just passed with "..."

File: R/summaries.R
Comment (antagomir) [inline R/summaries.R]: same

File: R/summaries.R
Comment (antagomir) [inline R/summaries.R]: hmm I guess the "name" argument should be also something that the users can customize, i.e. it should be available through the function arguments (either through ... passing if possible, or as explicit argument)

File: R/summaries.R
Comment (antagomir) [inline R/summaries.R]: this default should be given in the function call already, to make it explicit.

however, no rounding might be a good default; then the manpage example could additionally show how to do rounding to other digits,

**Review by antagomir — CHANGES_REQUESTED** [review]

**Comment by antagomir** [comment]
up

**Comment by antagomir** [comment]
@himmil this PR has been open for a month; could we aim to close soon?

**Comment by antagomir** [comment]
This PR has been merged but it says above that "This branch is out-of-date with the base branch", and this PR also hasnt been closed. Can we do that as well, or is something else still missing?

**Comment by himmil** [comment]
Apparently the testing file needs to be edited as well once the output table has changed.

**Comment by antagomir** [comment]
Yes, update all relevant files and complete the PR. Ask for support through Slack where necessary.

**Comment by antagomir** [comment]
Resolve the conversation issues when they become solved. This way we can remove points that have been successfully addressed.

**Comment by antagomir** [comment]
This branch is out-of-date with the base branch

**Comment by antagomir** [comment]
Thanks - one more request.

Could you also deprecate the name `countDominantTaxa` and take into use new name `countDominantFeatures`?

You can check other deprecated functions for an example.

The unit tests and vignette should be updated accordingly, after merging also the OMA examples.

**Review by antagomir — APPROVED** [review]

**Comment by antagomir** [comment]
check how to complete..

File: R/summaries.R
Comment (TuomasBorman) [inline R/summaries.R]: data[[name]] or data[, name]

File: R/summaries.R
Comment (TuomasBorman) [inline R/summaries.R]: Remove this, it's better to be exactly what user wants

File: R/summaries.R
Comment (TuomasBorman) [inline R/summaries.R]: Change to --> 

if( is.character(name) && length(name) == 1 )

Always use FALSE and not F. F is just a variable that has default value of FALSE (user can change the value of F to something else and then problems occur).

File: R/summaries.R
Comment (TuomasBorman) [reply R/summaries.R]: More precise

**Review by TuomasBorman — CHANGES_REQUESTED** [review]
Looks good, check comments; just minor things

**Comment by antagomir** [comment]
still on the way?

File: R/summaries.R
Comment (himmil) [reply R/summaries.R]: Needed to make some other changes instead, because column names with spaces caused problems

**Comment by himmil** [comment]
I also added the changes regarding to issues #393 and #377 into this PR, apologies for the inconvenience..

**Comment by antagomir** [comment]
For clarity, perhaps you can announce when this PR is ready for review.

**Comment by himmil** [comment]
It's ready for review.

**Comment by antagomir** [comment]
ok - @TuomasBorman

File: R/summaries.R
Comment (TuomasBorman) [inline R/summaries.R]: Remove this nested if --> 


    if(!.is_non_empty_string(name)){
        stop("'name' must be a non-empty single character value.",
             call. = FALSE)
    }

File: R/summaries.R
Comment (TuomasBorman) [inline R/summaries.R]: This removes the problem of spaces in column  names? --> Good!

One suggestion:

Do not feed name parameter to this function; it makes things little bit harder to read.

Remove these modifications regarding the naming of column, instead, store the table to variable and then change the column name.
```

        data$dominant_taxa <- dominant_taxa
        tab <- .tally_col_data()
        # Change the column name to correct one
```

File: R/summaries.R
Comment (TuomasBorman) [inline R/summaries.R]: THis line can be removed if you modify line
`
.tally_col_data(data, group, digits, colname, name)`

to 
`
.tally_col_data(data, group, digits, colname, name, ...)`

AND line

`.tally_col_data <- function(data, group, digits, colname, name){`

to
`
.tally_col_data <- function(data, group, digits = NULL, colname, name, ...){`

Your solution works, but for consistency (we have used this latter approach in other functions)

File: R/summaries.R
Comment (TuomasBorman) [inline R/summaries.R:320]: You could add a check for `digits`. It must be an integer or NULL

File: R/summaries.R
Comment (TuomasBorman) [inline R/summaries.R]: @antagomir SHould the `rel.freq` be `rel_freq`? By default, taxa column is named `dominant_taxa` (with underscore)

Or then we could change `dominant_taxa` to `dominant.taxa`. I think we do not have naming convention to tables

**Review by TuomasBorman — CHANGES_REQUESTED** [review]

File: R/summaries.R
Comment (himmil) [reply R/summaries.R]: Just letting you know that I've decided column name defaults to be with underscore and PR is ready for review if you agree

Comment (antagomir) [reply R/summaries.R]: nice!

**Review by antagomir — APPROVED** [review]

**Comment by antagomir** [comment]
Hmm seems merging is blocked until @TuomasBorman approves too. 

(we could see if it is possible to remove the _requirement_ of multiple approvals)

**Review by TuomasBorman — APPROVED** [review]

**Comment by himmil** [comment]
I don't have write access so could you either merge it for me or give me the permission?

**Pull request merged**
